buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersAware.java

```diff
@@ -34,6 +34,7 @@ default void useCluster(ElasticsearchCluster cluster) {
         }
 
         cluster.getNodes().stream().flatMap(node -> node.getDistributions().stream()).forEach(distro -> dependsOn(distro.getExtracted()));
+        cluster.getNodes().forEach(node -> dependsOn(node.getPluginAndModuleConfigurations()));
```

Comment:
I'm wondering if we should wrap `node.getPluginAndModuleConfigurations()` in a `Callable` to limit any odd configuration ordering stuff. That would allow folks to define a plugin/module dependency _after_ creating the test task and everything would still wire up properly.